pycrostates/cluster/aahc.py

```diff
@@ -0,0 +1,408 @@
+        tol: float = 1e-6,
+    ):
+        super().__init__()
+
+        self._n_clusters = _BaseCluster._check_n_clusters(n_clusters)
+        self._cluster_names = [str(k) for k in range(self.n_clusters)]
+
+        self._ignore_polarity = AAHCluster._check_ignore_polarity(
+            ignore_polarity
+        )
+        self._normalize_input = AAHCluster._check_ignore_polarity(
+            normalize_input
+        )
+        self._tol = AAHCluster._check_tol(tol)
+
+        # fit variables
+        self._GEV_ = None
+
+    def _repr_html_(self, caption=None):
+        from ..html_templates import repr_templates_env
+
+        template = repr_templates_env.get_template("AAHCluster.html.jinja")
+        if self.fitted:
+            n_samples = self._fitted_data.shape[-1]
+            ch_types, ch_counts = np.unique(
+                self.get_channel_types(), return_counts=True
+            )
+            ch_repr = [
+                f"{ch_count} {ch_type.upper()}"
+                for ch_type, ch_count in zip(ch_types, ch_counts)
+            ]
+            GEV = int(self._GEV_ * 100)
+        else:
+            n_samples = None
+            ch_repr = None
+            GEV = None
+
+        return template.render(
+            name=self.__class__.__name__,
+            n_clusters=self._n_clusters,
+            GEV=GEV,
+            cluster_names=self._cluster_names,
+            fitted=self._fitted,
+            n_samples=n_samples,
+            ch_repr=ch_repr,
+        )
+
+    @copy_doc(_BaseCluster.__eq__)
+    def __eq__(self, other: Any) -> bool:
+        """Equality == method."""
+        if isinstance(other, AAHCluster):
+            if not super().__eq__(other):
+                return False
+
+            attributes = (
+                "_ignore_polarity",
+                "_normalize_input",
+                "_tol",
+                "_GEV_",
+            )
+            for attribute in attributes:
+                try:
+                    attr1 = self.__getattribute__(attribute)
+                    attr2 = other.__getattribute__(attribute)
+                except AttributeError:
+                    return False
+                if attr1 != attr2:
+                    return False
+
+            return True
+        return False
+
+    @copy_doc(_BaseCluster.__ne__)
+    def __ne__(self, other: Any) -> bool:
+        """Different != method."""
+        return not self.__eq__(other)
+
+    @copy_doc(_BaseCluster._check_fit)
+    def _check_fit(self):
+        super()._check_fit()
+        # sanity-check
+        assert self.GEV_ is not None
+
+    @copy_doc(_BaseCluster.fit)
+    @fill_doc
+    def fit(
+        self,
+        inst: Union[BaseRaw, BaseEpochs],
+        picks: Picks = "eeg",
+        tmin: Optional[Union[int, float]] = None,
+        tmax: Optional[Union[int, float]] = None,
+        reject_by_annotation: bool = True,
+        n_jobs: int = 1,
+        *,
+        verbose: Optional[str] = None,
```

Comment:
@vferat `verbose` should probably be defined and set in the base-class. Which will remove from both ModKMeans and this class the `fill_doc` decorator around the `fit` method and its docstring.

Reply:
I agree with @mscheltienne . Should I implement this change in this PR?

Reply:
If you feel like it sure, else I'll do it after.